### PR TITLE
Clang format

### PR DESCRIPTION
**Pull request by Vikort**
Moved [pr](https://github.com/ostis-dev/sc-machine/pull/402)

### CODE REVIEW

**Review by KovalM — CHANGES_REQUESTED** [review]
Add check formatting to CI, script for only check formatting. Look to source PR, this changes are presented there

**Review by KovalM — CHANGES_REQUESTED** [review]
1. Add information about using code check to docs
2. Add setup pre-push hooks for code check. Add to docs info about install hooks

**Review by KovalM — APPROVED** [review]

File: scripts/ci/check-formatting.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+mkdir build -p
+pushd build
+cmake .. -DSC_CLANG_FORMAT_CODE=ON
+make clangformat_check
+popd
```
Comment (FallenChromium) [inline scripts/ci/check-formatting.sh:9]:
```suggestion
cmake -B build -DSC_CLANG_FORMAT_CODE=ON
cmake --build build --target clangformat_check
```
To fix the potential issue when `build` folder already exists

**Review by FallenChromium — CHANGES_REQUESTED** [review]
I see a major problem with this approach:
```shell
cd build
cmake ..
make
```
First, it changes the current directory which is undesirable. Second, it relies on `make` rather than `CMake` only, so if we're going to switch the build system someday (say `cmake -G Ninja`), our docs will be obsolete. I have a proposal: let's make this build command a standard:
```
cmake -B build
cmake --build build
```
It's shorter, only relies on CMake and it doesn't do the `cd`. If we'll adopt this approach in this PR, I'll make another one to redact it in the remaining places. I am not asking about this out of purely aesthetics reasons, look at what happened on my machine, for example:
image
I already have a build folder, but that shouldn't be a problem! If the script just used `build` as a target directory through cmake, rather than trying to create it manually, that would've passed.
I've also sent some minor notes about the documentation, but other than the problem stated above and docs changes, it looks good to me! The formatting rules themselves are a subject of discussion, personally I don't like how it looks, but it's very subjective so I'm not going to bring this up unless someone else wants to elaborate on the topic.

**Review by FallenChromium — APPROVED** [review]
Looks good to me! We could get rid of other `pushd`/`popd` or `cd`/`cd ..` logic too while we're at it, but it's unnecessary to do it in this PR.

**Pull request merged**